query.go

```diff
@@ -82,20 +91,21 @@ func (dht *IpfsDHT) runDisjointQueries(ctx context.Context, d int, target string
 
 	// distribute the shuffled K closest peers as seeds among the "d" disjoint queries
 	for i := 0; i < len(seedPeers); i++ {
-		queries[i%d].localPeers.Add(seedPeers[i])
+		queries[i%d].seedPeers = append(queries[i%d].seedPeers, seedPeers[i])
 	}
 
 	// start the "d"  disjoint queries
 	for i := 0; i < d; i++ {
 		query := queries[i]
 		go func() {
-			strictParallelismQuery(query)
+			query.runWithGreedyParallelism()
 			queryDone <- struct{}{}
 		}()
 	}
 
 loop:
 	// wait for all the "d" disjoint queries to complete before we return
+	// XXX: waiting until all queries are done is a security bug!!!
```

Comment:
Leaving this comment in the code without context will probably be confusing. Perhaps put in a couple line explanation like:

While running disjoint path queries may protect users from malicious peers they are directly connected to (assuming the peer is not eclipsed) it still allows the malicious peers to cause you to make many queries since we wait until all paths have completed.